homeassistant/components/thermostat/__init__.py

```diff
@@ -131,6 +133,22 @@ def state_attributes(self):
         if device_attr is not None:
             data.update(device_attr)
 
+        if hasattr(self, ATTR_MIN_TEMP):
```

Comment:
Instead of using `hasattr`, it would be better suited whem implemented as properties with default values.

```python
  @property
  def min_temp(self):
    self.hass.config.temperature(7, TEMP_CELCIUS)[0]

  def state_attributes(self):
    […]
    data[ATTR_MIN_TEMP] = self.min_temp
```

Any implementing thermostat would then just overwrite the min_temp property. This approach is in line with how other data is passed around within thermostats (for example `is_away_mode_on`)